src/com/facebook/buck/jvm/java/BaseCompileToJarStepFactory.java

```diff
@@ -98,6 +104,10 @@ public void createCompileToJarStep(
             classesToRemoveFromJar));
   }
 
+  Optional<String> getBootClasspath() {
```

Comment:
This feels like it should be abstract and return a non-optional String.  Do kotlin/groovy/scala not have this concept?  Can you document when and how a subclass should be overriding this?